### ISSUE TITLE
Kubernetes Hardening Guide

### ISSUE THREAD

**Issue by raesene** [kubernetes/sig-security#30]
**What would you like to be added**

As part of SIG-Security-Docs, we've been discussing the creation of a hardening guide for Kubernetes. We've got an initial document for the guide's creation here https://docs.google.com/document/d/1teb42X_c5_k8PNOSEEEbVnEr9aVAwWJXezBuf5fdmZU/edit

**Why is this needed**

The goal of the hardening guide is to provide guidance to cluster operators about how they can improve the security of their clusters. This will be done by discussing the major areas of security relating to a Kubernetes cluster, looking at the options available for hardening and the trade-offs inherent in them. In contrast to existing 3rd party documentation in this area (the CIS benchmark) which is a prescriptive audit style document, this guide should provide a more discursive approach.

**Table of Areas**

Section | Assignee | PR(s)
-- | -- | --
Threat Model | @cailynse |  
Control Plane Configuration |   |  
API Server Configuration |   |  
Scheduler Configuration |  @AnshumanTripathi   |  https://github.com/kubernetes/website/pull/45080
Controller Manager Configuration |   |  
File Permissions |   |  
Worker Node Configuration |   |  
PKI Management |   |  
Cluster Authentication | @raesene |  
Authorization | @bjornsen @vinayakankugoyal |  
Workload Security Configuration |   |  
Network Policy Configuration | @cailynse   |  
Resource Limits |   |  
Add-On Configuration |  |
etcd | @krol3 |

cc @savitharaghunathan @sftim

**Comment by sftim** [comment]
/remove-lifecycle rotten

Hope that's OK

**Comment by cailynse** [comment]
I'd be really interested in helping with this one!

**Comment by cailynse** [comment]
/remove-lifecycle rotten

@savitharaghunathan - is this something we could work on in the next cycle with SIG Security Docs?

**Comment by ericsmalling** [comment]
I'm interested in helping, even if just to help review and learn.

**Comment by cailynse** [comment]
Awesome! @ericsmalling feel free to pick up a section! I've just been researching and trying to fill in the TODOs from the top down!

**Comment by cailynse** [comment]
|Section|Assignee|PR(s)| 
|---|---|---|
|Threat Model|@cailynse|   | 
|Control Plane Configuration|   |   |
|API Server Configuration|   |   |
|Scheduler Configuration|   |   |
|Controller Manager Configuration|   |   |
|File Permissions|   |   |
|Worker Node Configuration|   |   |
|PKI Management|   |   |
|Cluster Authentication|   |   |
|Authorization|   |   |
|Workload Security Configuration|   |   |
|Network Policy Configuration|   |   |
|Resource Limits|   |   |
|Add-On Configuration|   |   |

**Comment by cailynse** [comment]
Threat Modelling PR: https://github.com/kubernetes/website/pull/39087

I'll also take Network Policy Configuration, please and thank you!

**Comment by bjornsen** [comment]
I'm interested in Authorization.

**Comment by raesene** [comment]
1st Draft for the Authentication section is open for comment on Hackmd https://hackmd.io/kxo4SRN3T3ipJHca2JNPTg

**Comment by raesene** [comment]
@bjornsen cool! I've added that assignment to the table at the top.

**Comment by vinayakankugoyal** [comment]
@bjornsen and me are going to be collaborating on Authorization.

**Comment by PushkarJ** [comment]
This might be of interest to the group here: https://github.com/cncf/tag-security/issues/1054

**Comment by raesene** [comment]
Hey all, so I've done some more fix-ups on the authentication section (https://hackmd.io/kxo4SRN3T3ipJHca2JNPTg?both) seems like it's probably in a decent enough spot to open a PR?

I know we had a quick chat about where in the docs site these pages should go, but I'm not sure we came to any firm conclusion.

**Comment by sftim** [comment]
> I think concepts -> security is a good home for the hardening guide

:+1:

A guide like this might then link to specific task pages, eg “Enable audit logging” “Configure KMS encryption for API objects”.

**Comment by reylejano** [comment]
> I think concepts -> security is a good home for the hardening guide. If needed we can create a folder and add items there. eg concepts -> security -> hardening guide -> auth mechanisms. @reylejano WDYT? should we bring this up in a sig-docs meeting or create a draft PR to get feedback on the content as well as the location?

I think concepts -> security -> hardening guide works which translates to`/docs/concepts/security/hardening-guide`

**Comment by sftim** [comment]
I wonder if we want to add a section to the list: cluster network security. That's different from network policy.

For example, you can harden Kubernetes by using MACsec between nodes, with appropriately frequent rekeying (lots of alternatives exist).

**Comment by raesene** [comment]
yeah I could see cluster network security being a good addition, as the topic is wider than just netpol.

On the organization of the page, that makes sense to me, I'll do a PR for the Authentication section with that structure and we can see if it works well :).

**Comment by raesene** [comment]
Opened the PR here https://github.com/kubernetes/website/pull/42486

**Comment by ashish493** [comment]
I've opened a PR for Authorization Guide - [kubernetes/website/#43623](https://github.com/kubernetes/website/pull/43623). 
Reviews and feedback are appreciated.

**Comment by ashish493** [comment]
I'm planning to pick the _Resource Limits_ topic  next. If anyone else is working on this topic, let me know if I can be of any help.

**Comment by slashben** [comment]
I am happy to work on
* Workload Security Configuration
* API server configuration

**Comment by AnshumanTripathi** [comment]
I volunteer to work on Scheduler Configuration.

**Comment by AnshumanTripathi** [comment]
First draft for hardening guide for scheduler configurations - https://hackmd.io/@anshumantripathi/By7z2BSsa

**Comment by AnshumanTripathi** [comment]
Question: should the hardening guides be moved to `References`?  This is with regards to question from @tengqm here
https://github.com/kubernetes/website/pull/45080#discussion_r1751203955

**Comment by sftim** [comment]
Guides are typically not reference material; see eg https://www.writethedocs.org/videos/eu/2017/the-four-kinds-of-documentation-and-why-you-need-to-understand-what-they-are-daniele-procida/ to see that these are different documentation kinds.

**Comment by AnshumanTripathi** [comment]
I will be picking hardening guide for resource limits.
